functions

```diff
@@ -0,0 +1,66 @@
+# DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
+ERROR='\033[0;31m'
+WARNING='\033[0;33m'
+INFO='\033[0m'
+DEFAULT='\033[0m'
+
+function _feedback() {
+  color="${1:-DEFAULT}"
+  case "${1}" in
+    ERROR)
+      >&2 echo -e "${!color}${1}:  ${2}${DEFAULT}" ;;
+    WARNING)
+      >&2 echo -e "${!color}${1}:  ${2}${DEFAULT}" ;;
+    *)
+      echo -e "${!color}${1}:  ${2}${DEFAULT}" ;;
+  esac
+}
+
+function process_security_tool_exit_status() {
+  exit_code="${1}"
+  security_tool="${2}"
+  action="${3}"
+
+  if [[ "${exit_code}" != 0 ]]; then
+    _feedback ERROR "Failed ${security_tool} ${action}"
+  else
+    _feedback INFO "Passed ${security_tool} ${action}"
+  fi
+
+  # If the shell is not interactive, skip multiple runs by timestamping the touch file
+  if [[ ! "${-}" =~ .*i.* ]]; then
+    date +%s > "/${security_tool}_complete"
+  fi
+
+  return ${exit_code}
+}
+
+# DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
+function terraform() {
+  security_tool="tfsec"
```

Comment:
Breaking out each of these into variables makes the script less readable to me, and it's hard to follow what exactly is being called. The function is called `terraform()`, so I don't see any issue putting `tfsec`  and its arguments in-line.

Reply:
Do you mean move that logic to the `make generate` so the resulting function is more readable?  That makes sense to me.